yt/data_objects/construction_data_containers.py

```diff
@@ -2109,3 +2112,294 @@ def _upload_to_sketchfab(self, data, files):
             mylog.error("Problem uploading.")
 
         return model_uid
+
+class YTOctree(YTSelectionContainer3D):
+    """A 3D region with all the data filled into an octree.
+
+    Parameters
+    ----------
+    right_edge : array_like
+        The right edge of the region to be extracted.  Specify units by supplying
+        a YTArray, otherwise code length units are assumed.
+    left_edge : array_like
+        The left edge of the region to be extracted.  Specify units by supplying
+        a YTArray, otherwise code length units are assumed.
+    n_ref: int
+        This is the maximum number of particles per leaf in the resulting
+        octree.
+    over_refine_factor: int
+        Each leaf has a number of cells, the number of cells is equal to
+        2**(3*over_refine_factor)
+    density_factor: int
+        This tells the tree that each node must divide into
+        2**(3*density_factor) children if it contains more particles than n_ref
+    ptypes: list
+        This is the type of particles to include when building the tree. This
+        will default to all particles
+
+    Examples
+    --------
+
+    octree = ds.octree(n_ref=64, over_refine_factor=2)
+    x_positions_of_cells = octree['x']
+    y_positions_of_cells = octree['x']
+    z_positions_of_cells = octree['x']
```

Comment:
you're using `'x'` for all three, i'd also explicitly use the `'index'` field type